### ISSUE TITLE
Stuck in Razorpay gateway after successful transaction Android 

### ISSUE THREAD

**Issue by vetriselvi-i2i** [razorpay/razorpay-android-sample-app#232]
## Description

Making the payment with Card/Phonepe via Razorpay gateway.
Phonepe - After payment success Razorpay still in the same page (like again I've to make payment screen)
Card payment - Not even processing any cards (Both master & visa cards) OTP screen is not coming.

## Razorpay SDK Version :
com.razorpay:checkout:1.6.6

## Android OS Version :
Some particular users only having this issue and they using Android 8 - Honor 9i mobile.

### What you did:

Trying to make payment from Phonepe via Razorpay gateway. 

### What happened:

After successfully completed the transaction in Phonepe, Razorpay is still staying in the pay now screen.

## Steps To Reproduce
Provide a detailed list of steps that reproduce the issue.

1. Choose Razorpay -> UPI -> Phonepe -> {pay in phonepe} -> success - automatically coming back to Razorpay from Phonepe -> app is still in the same page. (Sometimes Razorpay gateway closed but no callback received to my application)
2. Choose Razorpay -> Card -> enter card details -> I'm stuck there -> Not navigating to OTP screen.

### Suggested solution:

As per my findings Razorpay not giving a callback to my application.

**Comment by vivekshindhe** [comment]
@vetriselvi-i2i can you please share a sample project where you are able to reproduce the issue? That would help us a lot. Also v1.6.7 is released. You can try with that.

**Comment by vetriselvi-i2i** [comment]
@vivekshindhe Sure I'll update the new version and try now. Regarding the sample project to reproduce, I don't have one now. It's happening for some of the users only. Even I couldn't recreate this on my device, but few users keep on getting the same issue with the same flow.

**Comment by vivekshindhe** [comment]
Its most reproducible in Android 8 right? We'll try to reproduce the issue. But we'll need a bit more help from you on this as well.

**Comment by vetriselvi-i2i** [comment]
@vivekshindhe 
So the steps are below:
1. Just make a payment from Phonepe via RazorPay. Once payment is done Phonepe will close automatically and We'll be still staying in RazorPay screen (expected result: Razorpay should get close and it should give a callback to the application)
2. Choose a card option in RazorPay, by proceeding with it, it's not going to the OTP screen. (tried with both Master and Visa cards) 

Both above scenarios are for few users only, for others it's working in a positive flow. 
If you need more info please let me know.

**Comment by vivekshindhe** [comment]
I would need info of the few users this is happening to sir. Cause, we aren't able to reproduce these issues.

**Comment by vivekshindhe** [comment]
Android version, manufacturer, model.

**Comment by vetriselvi-i2i** [comment]
> Android version, manufacturer, model.
![IMG_20210528_213620](https://user-images.githubusercontent.com/54939814/120012305-f5b6e500-bffc-11eb-80fe-aa450d50c89a.jpg)
Hope this will help you @vivekshindhe

**Comment by vivekshindhe** [comment]
@vetriselvi-i2i Didn't realise you had access to the device where this is happening. Though the error is clear could you also please help with a video of it?

**Comment by vetriselvi-i2i** [comment]
@vivekshindhe Please find the vide here : Link: https://drive.google.com/file/d/1-YEhJe_STvaKdifw4C3T7nYgNkvcE-2Y/view

**Comment by vetriselvi-i2i** [comment]
> @vivekshindhe Please find the vide here : Link: https://drive.google.com/file/d/1-YEhJe_STvaKdifw4C3T7nYgNkvcE-2Y/view

@vivekshindhe Any updates? Is my info helpful for you?

**Comment by vivekshindhe** [comment]
@vetriselvi-i2i Yes, thank you for that. Trying to reproduce this on our end. Will update you once we have something on the same. Thank you for your patience.

**Comment by vivekshindhe** [comment]
@vetriselvi-i2i Not able to reproduce the issue on my device. Could you please send your merchant ID at vivek.shindhe@razorpay.com?

**Comment by vivekshindhe** [comment]
@vetriselvi-i2i closing the ticket due to inactivity. Please feel free to raise another issue or comment here for more support. Thank you.

**Comment by vetriselvi-i2i** [comment]
@vivekshindhe I've dropped a mail as you mentioned above. I also send follow-up mail. but I didn't receive any update from you.

**Comment by vivekshindhe** [comment]
Due to not being able to reproduce this on our end after several attempts we left this as intermittent and localized. In any occasion please update to version 1.6.15 and check. It should be resolved.thank you

**Comment by vetriselvi-i2i** [comment]
> Due to not being able to reproduce this on our end after several attempts we left this as intermittent and localized. In any occasion please update to version 1.6.15 and check. It should be resolved.thank you

Ok Thanks. Will check with the latest update.
